test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -10860,43 +10854,40 @@ def apply_config(self, config=None, validate=True, path=None):
 
         reconfig_time = int(time.time())
         try:
-            (fd, fn) = self.du.mkstemp()
+            fn = self.du.create_temp_file()
+            fd = open(fn, "w")
             for k in self._config_order:
                 if k in config:
                     if k in self._sched_config_comments:
-                        os.write(fd, "\n".join(self._sched_config_comments[k]))
-                        os.write(fd, "\n")
+                        fd.write("\n".join(self._sched_config_comments[k]))
+                        fd.write("\n")
                     v = config[k]
                     if isinstance(v, list):
                         for val in v:
-                            os.write(fd, k + ": " + str(val) + "\n")
+                            fd.write(k + ": " + str(val) + "\n")
                     else:
-                        os.write(fd, k + ": " + str(v) + "\n")
+                        fd.write(k + ": " + str(v) + "\n")
                 elif k in self._sched_config_comments:
-                    os.write(fd, "\n".join(self._sched_config_comments[k]))
-                    os.write(fd, "\n")
+                    fd.write("\n".join(self._sched_config_comments[k]))
+                    fd.write("\n")
             for k, v in self.sched_config.items():
                 if k not in self._config_order:
-                    os.write(fd, k + ": " + str(v).strip() + "\n")
+                    fd.write(k + ": " + str(v).strip() + "\n")
 
             if 'PTL_SCHED_CONFIG_TAIL' in self._sched_config_comments:
-                os.write(fd, "\n".join(
+                fd.write("\n".join(
                     self._sched_config_comments['PTL_SCHED_CONFIG_TAIL']))
-                os.write(fd, "\n")
-            os.close(fd)
+                fd.write("\n")
+            fd.close()
 
             if path is None:
-                sp = self.sched_config_file
-                if self.du.is_localhost(self.hostname):
-                    self.du.run_copy(self.hostname, sp, sp + '.bak', sudo=True)
-                else:
-                    cmd = ['mv', sp, sp + '.bak']
-                    self.du.run_cmd(self.hostname, cmd, sudo=True)
+                sp = os.path.join(self.pbs_conf['PBS_HOME'], "sched_priv",
+                                  "sched_config")
```

Comment:
This isn't multi-sched friendly.  You need to check for the sched_priv attribute first, and if it doesn't exist, use this join()